services/github/auth/acceptor.js

```diff
@@ -23,61 +24,64 @@ function setRoutes({ server, authHelper, onTokenAccepted }) {
     end('')
   })
 
-  server.route(/^\/github-auth\/done$/, (data, match, end, ask) => {
+  server.route(/^\/github-auth\/done$/, async (data, match, end, ask) => {
     if (!data.code) {
       log.log(`GitHub OAuth data: ${JSON.stringify(data)}`)
       return end('GitHub OAuth authentication failed to provide a code.')
     }
 
     const options = {
-      url: 'https://github.com/login/oauth/access_token',
       method: 'POST',
       headers: {
         'Content-type': 'application/x-www-form-urlencoded;charset=UTF-8',
-        'User-Agent': userAgent,
       },
-      form: queryString.stringify({
+      form: {
         // TODO The `_user` and `_pass` properties bypass security checks in
         // AuthHelper (e.g: enforceStrictSsl and shouldAuthenticateRequest).
         // Do not use them elsewhere. It would be better to clean
         // this up so it's not setting a bad example.
         client_id: authHelper._user,
         client_secret: authHelper._pass,
         code: data.code,
-      }),
+      },
```

Comment:
guess `got` will handle the query param mappings for us?

Reply:
Yep. By default got will use URL encoding for the `form` object. It will also set the content-type header to `application/x-www-form-urlencoded`, but I left the custom header with the `;charset=UTF-8` in place.

The "a code is provided" test case decodes the request body with `queryString.parse()` to make assertions about it, so that behaviour is covered under test.